mail/mail-server/Portfile

```diff
@@ -72,40 +72,12 @@ use_configure           no
 
 pre-build {
     # require these port dependency variants
-    # MacPorts doesn't support variant dependencies, so do this this by hand
-    # See: https://trac.macports.org/ticket/126
-    # https://lists.macports.org/pipermail/macports-users/2015-September/039201.html
-
-    set required_variants_flag true
-    set required_variants_message ""
-
-    if { [catch {set result [registry_active postfix]}]
-        || [lindex [lindex ${result} 0] 3] \
-            ne "+[join ${postfix_required_variants} +]" } {
-        append required_variants_message "\
-Postfix not installed with required variants. Please install:
-
-    sudo port -pN install postfix +[join ${postfix_required_variants} +]
-
-"
-        set required_variants_flag false
+    foreach var ${postfix_required_variants} {
+        require_active_variants postfix ${var}
     }
-    if { [catch {set result [registry_active dovecot]}]
-        || [lindex [lindex ${result} 0] 3] \
-            ne "+[join ${dovecot_required_variants} +]" } {
-        append required_variants_message "\
-Dovecot not installed with required variants. Please install:
-
-    sudo port -pN install dovecot +[join ${dovecot_required_variants} +] \[+apns\]
 
-"
-        set required_variants_flag false
-    }
-    if {${required_variants_flag} ne true} {
-        ui_error ${required_variants_message}
-        # one of these will exit with error if the ports are not installed at all
-        registry_active postfix
-        registry_active dovecot
+    foreach var ${dovecot_required_variants} {
+        require_active_variants dovecot ${var}
     }
 }
```

Comment:
this doesn't need to be in a `pre-build` block and could/should just be:
```
require_active_variants postfix dovecot_sasl pcre smtputf8 tls
require_active_variants dovecot solr
```

no need to define new variables and loop over them.
```

Reply:
Thanks, done. I kept the variable because it's in a `notes` string.